core/config/config.go

```diff
@@ -20,201 +19,358 @@ import (
-// Operator has a different config and is meant to be used by the operator CLI.
-func NewConfig(ctx *cli.Context) (*Config, error) {
-	var configRaw ConfigRaw
+type OperatorConfig struct {
+	BaseConfig *BaseConfig
+	Operator   struct {
+		Address                   common.Address `yaml:"address"`
+		EarningsReceiverAddress   common.Address `yaml:"earnings_receiver_address"`
+		DelegationApproverAddress common.Address `yaml:"delegation_approver_address"`
+		StakerOptOutWindowBlocks  int            `yaml:"staker_opt_out_window_blocks"`
+		MetadataUrl               string         `yaml:"metadata_url"`
+		RegisterOperatorOnStartup bool           `yaml:"register_operator_on_startup"`
+	} `yaml:"operator"`
+}
 
-	configFilePath := ctx.GlobalString(ConfigFileFlag.Name)
-	if configFilePath != "" {
-		err := sdkutils.ReadYamlConfig(configFilePath, &configRaw)
-		if err != nil {
-			fmt.Println("Could not read yaml config file")
-			return nil, err
-		}
+var (
+	// Required Flags
+	BaseConfigFileFlag = cli.StringFlag{
+		Name:     "base-config-file",
+		Required: true,
+		Usage:    "Load base configurations from `FILE`",
 	}
+	AggregatorConfigFileFlag = cli.StringFlag{
+		Name:     "aggregator-config-file",
+		Required: true,
+		Usage:    "Load aggregator configurations from `FILE`",
+	}
+	OperatorConfigFileFlag = cli.StringFlag{
+		Name:     "operator-config-file",
+		Required: true,
+		Usage:    "Load operator configurations from `FILE`",
+	}
+	// Optional Flags
+)
+
+func NewAggregatorConfig(baseConfigFilePath, aggregatorConfigFilePath string) (*AggregatorConfig, error) {
+
+	if _, err := os.Stat(baseConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup base config file does not exist")
+	}
+
+	if _, err := os.Stat(aggregatorConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup aggregator config file does not exist")
+	}
+
+	baseConfig, err := newBaseConfig(baseConfigFilePath)
 
-	logger, err := sdklogging.NewZapLogger(configRaw.Environment)
 	if err != nil {
-		fmt.Println("Could not initialize logger")
+		log.Fatal("Error reading base config: ", err)
 	}
 
-	var alignedLayerDeploymentRaw AlignedLayerDeploymentRaw
-	alignedLayerDeploymentFilePath := ctx.GlobalString(AlignedLayerDeploymentFileFlag.Name)
-	if _, err := os.Stat(alignedLayerDeploymentFilePath); errors.Is(err, os.ErrNotExist) {
-		logger.Errorf("Path does not exist", "path", alignedLayerDeploymentFilePath)
-		return nil, err
+	var aggregatorConfigFromYaml AggregatorConfig
+	err = sdkutils.ReadYamlConfig(aggregatorConfigFilePath, &aggregatorConfigFromYaml)
+
+	if err != nil {
+		log.Fatal("Error reading aggregator config: ", err)
 	}
-	err = sdkutils.ReadJsonConfig(alignedLayerDeploymentFilePath, &alignedLayerDeploymentRaw)
+
+	return &AggregatorConfig{
+		BaseConfig: baseConfig,
+		Aggregator: aggregatorConfigFromYaml.Aggregator,
+	}, nil
+
+}
+
+func NewOperatorConfig(baseConfigFilePath, operatorConfigFilePath string) (*OperatorConfig, error) {
```

Comment:
If we are using log.Fatal instead of returning error, remove error from return type. Two options here:
1. Go back to return error, and log error instead of fatal (which kills process)
2. Return just *OperatorConfig